packages/theme/middleware/order-details.js

```diff
@@ -0,0 +1,13 @@
+import { orderGetters } from '@vue-storefront/spree';
+
+export default async ({ app, $vsf }) => {
+  const callFallbackRedirect = () => app.context.redirect('/my-account/order-history');
+
+  const { pageName, id: orderId } = app.context.route.params;
+  if (pageName !== 'order-details') return;
+  if (!orderId) return callFallbackRedirect();
+
+  const orders = await $vsf.$spree.api.getOrders();
+  if (orders.some(order => orderGetters.getId(order)?.toLowerCase() === orderId)) return;
```

Comment:
This seems a bit too heavy - no point in pulling all orders if we want to check a single order. 
There's an endpoint for pulling a single order by number: https://api.spreecommerce.org/docs/api-v2/b3A6MTgwNTI4NjA-retrieve-an-order

We could add an endpoint in our api-client package to support it and then use it here.

Reply:
Actually, should we do a redirect if an order is not found? Maybe we should just let the customer enter the page and show a message "This order could not be found"?

Reply:
> This seems a bit too heavy - no point in pulling all orders if we want to check a single order. There's an endpoint for pulling a single order by number: https://api.spreecommerce.org/docs/api-v2/b3A6MTgwNTI4NjA-retrieve-an-order
> 
> We could add an endpoint in our api-client package to support it and then use it here.

I've added the getOrder API method and updated useUserOrder accordingly.

Reply:
> Actually, should we do a redirect if an order is not found? Maybe we should just let the customer enter the page and show a message "This order could not be found"?

I've changed the redirection logic into raising the 404 error (which displays the error.vue layout) in the middleware.